src/YoutubeAtom.tsx

```diff
@@ -227,13 +227,20 @@ export const YoutubeAtom = ({
                       adTargeting.customParams,
                       consentState,
                   );
-        const embedConfig = encodeURIComponent(JSON.stringify({ adsConfig }));
-        const originString = origin
-            ? `&origin=${encodeURIComponent(origin)}`
-            : '';
-        setIframeSrc(
-            `https://www.youtube.com/embed/${assetId}?embed_config=${embedConfig}&enablejsapi=1&widgetid=1&modestbranding=1${originString}&autoplay=1`,
+
+        const params = new URLSearchParams({
+            ...(origin ? { origin } : undefined),
```

Comment:
I'm a little lost here if `origin` is `undefined` you are going to spread on `undefined` which is invalid? Interesting as TypeScript isn't saying so, so is it my understanding that's wrong?

See: https://www.typescriptlang.org/play?#code/MYewdgzgLgBFBcMBKBTUAnAJgHmuglmAOYA0MehRAfDAD4wCuYmKAZoSpjALyPNsdMAKFCRYRHjADeMAHTyAFLAD8cGIiYt2YTgEoYAXyEjwEEABsUs8yCJLdQA

Reply:
Aah… weird it’s not picking it up. Fixed by spreading to `{}`.